### PR TITLE
Revise Compiler backend API

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
This revised Compiler backend API to have `getToolchainTypes()` and
`getToolchains()` methods, which are deprecating previous `toolchains()`
method.

For #573

ONE-vscode-DCO-1.0-Signed-off-by: Hyun Sik Yoon <hyunsik.yoon.1024@gmail.com>

### CODE REVIEW

File: src/Backend/Compiler.ts [hunk 1]
```diff
@@ -28,9 +28,35 @@ import {Toolchains} from './Toolchain';
...
+   * in such case, we can call those two.
+   *    const first20 = getToolchains('nightly', 0, 20);
+   *    const next30 = getToolchains('nightly', 20, 30);
+   *
+   * @param toolchainType One of value returned from toolchainTypes()
+   * @param start starting index of whole installable toolchains sorted by version,
+   *              recent version first. 0-based index.
+   * @param count number of Toolchain returned
+   *
+   * @throw Error when toolchainTypes are not supported
```
[the remark this answers is not in the document]
Comment (hyunsik-yoon) [reply src/Backend/Compiler.ts:57]: Comment regarding such situation was added.  😸

**Review by yunjayh — APPROVED** [review]
LGTM

**Review by llFreetimell — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**